### DISCUSSION TITLE
Response caching

### DISCUSSION THREAD

**Discussion by bobneuman** [PyGithub/PyGithub#2997] [Ideas]
We're using pyGithub to deploy issues created from templates and use various validation before creating the content.  It works for small quantity of issues, but rapidly runs into the API rate limits.  I'm migrating over to our own grapdhql calls to implement caching with timeouts for those functions used to deploy the issues.  

 Have you considered adding response caching with timeouts for queries?   pyGithub does call a rate limited API.

**Comment by EnricoMi** [comment]
Would lazy PyGithub objects help? See #2746.
